tensortrade/exchanges/simulated/simulated_exchange.py

```diff
@@ -146,67 +162,106 @@ def transform_data_frame(self) -> bool:
+            self._trades = self._trades.append({'step': self._current_step,
+                                                'symbol': trade.symbol,
+                                                'type': trade.trade_type,
+                                                'amount': trade.transact_amount,
+                                                'price': trade.transact_total}, ignore_index=True)
 
         self._portfolio[self._base_instrument] = self._balance
 
-        self._performance = self._performance.append({
-            'balance': self.balance,
-            'net_worth': self.net_worth,
-        }, ignore_index=True)
+        log.update({'balance': self.balance,
+                    'net_worth': self.net_worth
+                    })
+        self._performance = self._performance.append( log , ignore_index=True)
 
     def execute_trade(self, trade: Trade) -> Trade:
-        current_price = self.current_price(symbol=trade.symbol)
-        commission = self._commission_percent / 100
-        filled_trade = trade.copy()
-
-        if filled_trade.is_hold or not self._is_valid_trade(filled_trade):
-            filled_trade.amount = 0
+        trade = trade.copy()
+        transact_amount = trade.order_amount
+        transact_price = trade.order_price
 
-            return filled_trade
+        if trade.is_hold:
+            # hodl boyz!
+            trade.transact_amount = transact_amount
+            trade.transact_price = transact_price
+        else:
+            trade.order_commission = transact_price * self.commission_percent
 
-        if filled_trade.is_buy:
-            price_adjustment = (1 + commission)
-            filled_trade.price = max(round(current_price * price_adjustment,
-                                           self._base_precision), self.base_precision)
-            filled_trade.amount = round(
-                (filled_trade.price * filled_trade.amount) / filled_trade.price, self._instrument_precision)
-        elif filled_trade.is_sell:
-            price_adjustment = (1 - commission)
-            filled_trade.price = round(current_price * price_adjustment, self._base_precision)
-            filled_trade.amount = round(filled_trade.amount, self._instrument_precision)
-
-        filled_trade = self._slippage_model.fill_order(filled_trade, current_price)
-
-        self._update_account(filled_trade)
+        if trade.is_buy:
+            if trade.is_limit_buy:
+                # our buying power will fluxuate slightly as the market moves
+                slip_adjusted_amount = (self.balance * (1-(self._max_amount_slippage_percent/100)) * transact_amount) / transact_price
+                transact_amount = self._amount_slip_model.random_slip(slip_adjusted_amount)
+            elif trade.is_market_buy:
+                # the price can fluxuate up or down during the market buy order
+                transact_price = self._price_slip_model.random_slip(transact_price)
+            else:
```

Comment:
Is the catch all necessary? A trade should likely not be executed if it has an invalid trade type.

Reply:
This exchange currently only accepts `MARKET` and `LIMIT` buys/sells, so it makes sense to fail if another type of buy/sell is transacted.